### PR TITLE
Allow collections to be ordered by publish date

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Add ability to sort collections by publish date.
 - `/collections` - applies default ordering, yet to be defined
 - `/collections?order_by=publish_date` - collections ordered by publish date
 - `/collections?order_by=notrecognised` - 400 response and error body

### How to review
Review changes / tests / component tests

### Who can review
Anyone

### CODE REVIEW

**Review by s-xton — DISMISSED** [review]

File: api/collection_test.go [hunk 1]
```diff
@@ -151,6 +203,52 @@ func TestGetCollections_collectionStoreError(t *testing.T) {
...
+				response := models.ErrorsResponse{}
+				err = json.Unmarshal(body, &response)
+				So(err, ShouldBeNil)
+				So(len(response.Errors), ShouldEqual, 1)
+				So(response.Errors[0].Message, ShouldEqual, collections.ErrInvalidOrderBy.Error())
+			})
+		})
+	})
+}
+
```
Comment (daiLlew) [inline api/collection_test.go:251]: Can you add a test to cover order empty/not supplied. I can see that is covered in to the code but it would be good to prove it with a test case.

File: collections/ordering.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package collections
+
+import "errors"
+
+var ErrInvalidOrderBy = errors.New("invalid order_by")
```
Comment (daiLlew) [inline collections/ordering.go:5]: Following on from the linter discussion today - should probably add GoDoc for public types.

**Review by s-xton — APPROVED** [review]
